checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -4,89 +4,369 @@
+     * @return whether the caller can skip the receiver subtype check.
+     */
+    @Override
+    protected boolean skipReceiverSubtypeCheck(MethodInvocationTree node,
+            AnnotatedTypeMirror methodDefinitionReceiver,
+            AnnotatedTypeMirror methodCallReceiver) {
+
+        AnnotationMirror primaryGb = methodCallReceiver.getAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+        AnnotationMirror effectiveGb = methodCallReceiver.getEffectiveAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+
+        // If the receiver actual parameter has type @GuardSatisfied, skip the subtype check.
+        // Consider only a @GuardSatisfied primary annotation - hence use primaryGb instead of effectiveGb.
+        if (primaryGb != null && AnnotationUtils.areSameByClass(primaryGb, checkerGuardSatisfiedClass)){
+            AnnotationMirror primaryGbOnMethodDefinition = methodDefinitionReceiver.getAnnotationInHierarchy(GUARDEDBYUNKNOWN);
+            if (primaryGbOnMethodDefinition != null && AnnotationUtils.areSameByClass(primaryGbOnMethodDefinition, checkerGuardSatisfiedClass)) {
+                return true;
+            }
+        }
+
+        if (AnnotationUtils.areSameByClass(effectiveGb, checkerGuardedByClass)) {
+            Set<AnnotationMirror> annos = methodDefinitionReceiver.getAnnotations();
+            for (AnnotationMirror anno : annos) {
+                if (AnnotationUtils.areSameByClass(anno, checkerGuardSatisfiedClass)) {
+                    MethodInvocationNode methodInvocationNode = (MethodInvocationNode) atypeFactory.getNodeForTree(node);
+                    Node receiverNode = methodInvocationNode.getTarget().getReceiver();
+                    checkPreconditions(node, receiverNode,
+                            generatePreconditionsBasedOnGuards(methodCallReceiver));
+
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    protected Set<? extends AnnotationMirror> getExceptionParameterLowerBoundAnnotations() {
+        Set<? extends AnnotationMirror> tops = atypeFactory.getQualifierHierarchy().getTopAnnotations();
+        Set<AnnotationMirror> annotationSet = AnnotationUtils.createAnnotationSet();
+        for (AnnotationMirror anno : tops) {
+            if (anno.equals(GUARDEDBYUNKNOWN)) {
+                annotationSet.add(GUARDEDBY);
+            } else {
+                annotationSet.add(anno);
+            }
+        }
+        return annotationSet;
+    }
+
+    /***
+     * Given an AnnotatedTypeMirror containing a @GuardedBy annotation, returns the set of lock expression preconditions
+     * specified in the @GuardedBy annotation.
+     * Returns an empty set if no such expressions are found.
+     *
+     * @param atm the AnnotatedTypeMirror containing the @GuardedBy annotation with the lock expression preconditions.
+     * @return a set of lock expression preconditions that can be processed by checkPreconditions.
```

Comment:
I agree a dedicated type would make it easier to understand. That said, quite a few methods use Set<Pair<String, String>> for this purpose:
ContractsUtils:
getPrecondition
getPreconditions
getPostcondition
getPostconditions

BaseTypeVisitor:
checkPostconditions
checkPostconditionsConsistency
checkPreconditions
checkPreconditionsConsistency
resolveContracts
OverrideChecker.checkPreAndPostConditions

I think if we change it for some of these, we should change it for all of them. Would you like me to make this change?